### PR TITLE
[PM-169][PM-142][PM-191] Add Environments to Web and Desktop

### PR DESCRIPTION
**Pull request by andrebispo5**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Add environment selector on web login/registration/trial initiation and desktop app.

## Code changes

- **apps/web/src/app/layouts/frontend-layout.component.html:** Since this is the component for the footer on unauthed pages, the code to have the env selector was added here. Links to EU and US servers are hardcoded.

- **apps/desktop/src/app/components/environment-selector.component.ts:** This component holds the logic to the environment selector for the desktop app. The logic here might be converted to the lib core if the code could be reused for the browser app.

## Screenshots

**Web:**
image
image
image
**Desktop:**
image
image
image
image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by differsthecat — COMMENTED** [review]
The commits and diff on this are a little strange; it looks like you merged master into this branch, but not the base feature branch. You're probably on top of that as I type this 😅

**Comment by JaredSnider-Bitwarden** [comment]
Minor question:  is the difference in positioning of the env selector menu between web and desktop intentional?
Web:
image
vs
Desktop:
image

File: apps/desktop/src/app/components/environment-selector.component.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  async toggle(option: ServerEnvironment) {
+    this.isOpen = !this.isOpen;
+    if (option === ServerEnvironment.EU) {
+      await this.environmentService.setUrls({ base: "https://vault.bitwarden.eu" });
+    } else if (option === ServerEnvironment.US) {
+      await this.environmentService.setUrls({ base: "https://vault.bitwarden.com" });
+    } else if (option === ServerEnvironment.SelfHosted) {
+      this.settings();
```
Comment (JaredSnider-Bitwarden) [inline apps/desktop/src/app/components/environment-selector.component.ts:80]: Minor naming nitpick:  I feel like this name should be something more clear like `openSelfHostedSettingsModal(...)` vs just `settings(...)`

File: apps/desktop/src/app/components/environment-selector.component.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import { EnvironmentService } from "@bitwarden/common/abstractions/environment.service";
+import { FeatureFlag } from "@bitwarden/common/enums/feature-flag.enum";
+
+import { EnvironmentComponent } from "../../auth/environment.component";
+
+@Component({
+  selector: "environment-selector",
+  templateUrl: "environment-selector.component.html",
+  animations: [
+    trigger("transformPanel", [
```
Comment (JaredSnider-Bitwarden) [inline apps/desktop/src/app/components/environment-selector.component.ts:17]: Reminder from our discussion: check if you are using these.

Comment (andrebispo5) [reply apps/desktop/src/app/components/environment-selector.component.ts:17]: Double checked and it's being used. https://github.com/bitwarden/clients/pull/5294/files#diff-df5df8a01f4c721f02610b20f015bf15ca99289bd6fd05d3892d1873b81f144fR29

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Great work on this!

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Great work on this!

**Pull request merged**